### ISSUE TITLE
Enums do not work in 2.0 if they are defined as objects

### ISSUE THREAD

**Issue by hsn10** [outworkers/phantom#607]
fails to compile due to error resulted from macro expansion:

`object E1 extends Enumeration {

     type E1 = Value
     val Mon, Tue = Value
}
--cut--
import E1.E1
case class  Row(
  e: E1
)
class CRow extends CassandraTable[ConcreteRow, Row] {
   object e extends EnumColumn [E1] (this)
}
`

If I define enums like you do as trait E1 extends Enumeration and then use E1#Value as type then it works.
I never heard about # separator in scala. As far i know everybody defines enums as objects, not traits.

**Comment by alexflav23** [comment]
Hi @hsn10,

The `#` is a reference to an abstract type member and you should use the `EnumColumn[E1#Value]` syntax. This is because the type `E1` is a singleton type, `E1.type` is the type of the anonymous class type that gets created.

You don't need to define enums as traits first, you do need to pass in the value type though.

Regards,
Flavian

**Comment by hsn10** [comment]
If i modify code to:

`

import E1
case class Row(
e: E1.type#Value
)
class CRow extends CassandraTable[ConcreteRow, Row] {
object e extends EnumColumn [E1.type#Value] (this)
}
`

Then it does not work because Value is protected method. Can you provide an example of working non trait based enum?

**Comment by alexflav23** [comment]
Hi @hsn10,

I see what you mean, we are actively working to fix this, it will take a week or two for the fix to get deployed but it will land shortly, I will keep you up to date.

Regards,

**Comment by alexflav23** [comment]
Hi @hsn10,

Fixed in `2.1.1`, will be on Maven Central in 2 hours. Simply use the `object.Value` method and you will get what you want.

```scala
object Enum extends Enumeration {
  val one = Value("one")
  val two = Value("two")
}

class MyTable extends CassandraTable[MyTable, Rec] {
   ...
   object enum extends EnumColumn[Enum.Value](this)
}
```
